### PR TITLE
[WIP] Rework radio documentation

### PR DESCRIPTION
**Pull request by blikblum**
The current Radio documentation is bit confusing. First it show an full example using Object radio integration features than goes back to explain Radio concepts, keeping going back and forth to radio integration. 

### Proposed changes

 - Reorganize the document using the following structure
```
Introduction
Index
Radio Concepts
  Channel
  Events
  Requests
Marionette Integration
   API
   Examples
```
 - Document once method in event API
 - Make clear that an inline function can be used as event / request handler
 - Removes redundant info

It is still WIP. Needs to update the index and check the lint rules.
Creating now to get initial feedback

### CODE REVIEW

File: docs/backbone.radio.md [hunk 1]
```diff
@@ -85,7 +23,9 @@ access to this API.
   * [Listening to Requests on Objects](#listening-to-requests-on-objects)
   * [When to use Requests](#when-to-use-requests)
 
-## Channel
+## Radio Concepts
+
```
Comment (scott-w) [inline docs/backbone.radio.md]: I'd add a little text here that says:

```markdown
The `Radio` message bus exposes some core concepts:

* `Channel` - a namespace mechanism.
* `Request` - execute single functions in a different part of your application.
* `Event` - alert other parts of your application that something happened.
```

It'll give the reader an idea of the contents of the 'chapter' and the scope of what they're about to read.

Comment (blikblum) [inline docs/backbone.radio.md]: Done.
I also updated the index and fixed the lint issues. By my side is ready to go

**Comment by paulfalgout** [comment]
:+1: LGTM

**Comment by blikblum** [comment]
I'm ok with it

**Pull request merged**